File: angular/src/app/identity/identity.model.ts
import { Account } from './account.model';
import { Profile } from './profile.model';

export interface Identity {
  account: {
    connectedAccounts: Account[];
    profile: Profile;
    properties: {
      subject: string;
    };
  };
  scopes: string[];
  sandbox: boolean;
}

File: src/main/java/com/dnastack/ddap/ic/account/model/IdentityModel.java
package com.dnastack.ddap.ic.account.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@Data
public class IdentityModel {

    private Object account;
    private List<String> scopes;
    private List<Access> accesses;
    private boolean sandbox;

    @Data
    public static class Access {

        private Target target;
        private Boolean isAdmin;

    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Target {
        private Service service;
        private String id;
    }

    public enum Service {
        DAM, IC;
    }
}
